woocommerce/woocommerce-ios
# [Mobile Payments] Pair to reader when collecting a payment and a reader is not paired

Closes #4062

⚠️ This PR is against the feature branch implementing support for Card Present Payments, not against develop ⚠️

Present the UI to discover and connect to card readers (v2) when an attempt to collect a payment is initiated without being connected to a reader

## Changes
* OrderDetailsViewModel will check if there is a reader connected
* If there is no reader connected, OrderDetailsViewController will present the UI to discover and connect.
* Once we have connected to a reader, the UI for connection is dismissed, and we initiate the process to collect the payment.

## How to test
Testing requires:

1. Setting up a store for Card Present Payment Testing (also P91TBi-4BH-p2 for more specific instructions)
2. WCPay 2.4.0

* Checkout the branch
* Navigate to an order eligible for in-person payments, without connecting to a reader.
* Tap "Collect payment". There should be a fullscreen modal presenting the UI to discover and connect. Once the process is completed, the process to collect a payment should be initiated.
* Navigate to a different eligible order. Now there there is a reader connected, tapping "Collect payment" should start the flow to collect the payment directly.

Update release notes:
